### PR TITLE
CLOUDP-69557: reduce datalake e2e flakyness

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-69557

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Trying to move the list later to give it time to show something and hopefully stop the flakyness

### CODE REVIEW

File: e2e/atlas/data_lake_test.go [hunk 1]
```diff
@@ -63,24 +63,24 @@ func TestDataLakes(t *testing.T) {
...
+		if err = json.Unmarshal(resp, &dataLake); a.NoError(err) {
+			a.Equal(dataLakeName, dataLake.Name)
 		}
 	})
 
 	t.Run("List", func(t *testing.T) {
 		cmd := exec.Command(cliPath,
 			atlasEntity,
-			clustersEntity,
+			datalakeEntity,
```
Comment (gssbzn) [inline e2e/atlas/data_lake_test.go:83]: 🤦

File: e2e/iam/project_api_keys_test.go [hunk 2]
```diff
@@ -51,11 +54,21 @@ func TestProjectAPIKeys(t *testing.T) {
...
+			a.Equal(desc, key.Desc)
 			ID = key.ID
 		}
 	})
 
+	if ID == "" {
+		assert.FailNow(t, "Failed to create API key")
+	}
+
+	defer func() {
```
Comment (gssbzn) [inline e2e/iam/project_api_keys_test.go:66]: so if you have a project API key it's still bound to the org even if removed from a project

File: internal/cli/iam/iam.go [hunk 3]
```diff
@@ -15,8 +15,8 @@
 package iam
 
 import (
-	"github.com/mongodb/mongocli/internal/cli/organizations"
-	"github.com/mongodb/mongocli/internal/cli/projects"
+	"github.com/mongodb/mongocli/internal/cli/iam/organizations"
+	"github.com/mongodb/mongocli/internal/cli/iam/projects"
```
Comment (gssbzn) [inline internal/cli/iam/iam.go:19]: after here is just moving the commands to a new package

File: build/ci/evergreen.yml [hunk 4]
```diff
@@ -521,6 +521,24 @@ tasks:
           MCLI_PUBLIC_API_KEY: ${atlas_public_api_key}
           MCLI_SERVICE: cloud
           E2E_TAGS: atlas,search
+  # Private endpoints can be flaky when multiple tests run concurrently so keeping this out of the PR suite
+  - name: atlas_private_endpoint_e2e
```
Comment (gssbzn) [inline build/ci/evergreen.yml:525]: the comment above says it all

File: e2e/iam/org_api_keys_test.go [hunk 5]
```diff
@@ -129,6 +136,10 @@ func TestOrgAPIKeys(t *testing.T) {
 		cmd.Env = os.Environ()
 		resp, err := cmd.CombinedOutput()
 
-		assert.NoError(t, err, string(resp))
+		a := assert.New(t)
+		if a.NoError(err, string(resp)) {
+			expected := fmt.Sprintf("API Key '%s' deleted\n", ID)
+			a.Equal(expected, string(resp))
+		}
```
Comment (andreaangiolillo) [inline e2e/iam/org_api_keys_test.go:143]: 💯

File: internal/cli/global_opts.go [hunk 6]
```diff
@@ -71,7 +71,7 @@ func (opts *GlobalOpts) PreRunE(cbs ...cmdOpt) error {
 // PreRunEOrg is a function to call before running the command,
 // this will validate the org ID and call any additional function pass as a callback
 func (opts *GlobalOpts) PreRunEOrg(cbs ...cmdOpt) error {
-	if opts.ConfigOrgID() == "" && opts.OrgID == "" {
+	if opts.ConfigOrgID() == "" {
 		return ErrMissingOrgID
```
Comment (andreaangiolillo) [inline internal/cli/global_opts.go:75]: 💯

**Pull request merged**